### PR TITLE
Manual turret aiming rework

### PR DESCRIPTION
**Pull request by Coolthulhu**
Quite big rework of manual turret aiming. Not completely finished (lacks good display), but already far better than the one before.

Turrets can be mounted by the player. This is done automatically when player without a weapon tries to fire while standing under a turret. This spawns a fake gun and allows it to be aimed and shot.
Player can also change the turret mode (burst/single) while standing under it. This is done when an unarmed player tries to change gun mode while in a vehicle with a turret at player's location.

Reworked aiming from controls completely. Turret aiming mode is automatically selected when a weaponless player tries to fire while standing over controls of a vehicle with at least one active turret that can be aimed.
All turrets are aimed at once and aiming range is the radius of a circle centered at player and big enough to cover all points that any turret could aim at.

As I mentioned, the display isn't perfect yet. Turrets without enough range will simply not be aimed and the player will only notice it when the turrets don't fire. Later on I intend to add redrawing of the aimed turrets in color (and/or symbol) depending on their ability to fire (can fire, no ammo, no range, turned off etc.). Though this probably deserves own PR, as this one is quite big already.
Also, currently turrets are hidden while the player is in the vehicle.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -3438,6 +3438,31 @@ void player::print_recoil( WINDOW *w ) const
     }
 }
 
+int player::draw_turret_aim( WINDOW *w, int line_number, const point &targ ) const
+{
+    (void)w;
```
Comment (BevapDin) [inline src/player.cpp]: You're using `w` and `targ`, no need to add this pseudo usage.

**Comment by Rivet-the-Zombie** [comment]
This looks awesome! Folks have been wanting the ability to manually fire mounted weapons for a long time.

Good stuff, Coolthulhu!

File: src/vehicle.cpp [hunk 2]
```diff
@@ -4980,105 +4981,155 @@ std::string aim_type( const vehicle_part &part )
...
+bool vehicle::aim_turrets()
+{
+    std::vector< int > turrets = all_parts_with_feature( "TURRET", true );
+    // We only want turrets that can aim by themselves and are on
+    auto cant_shoot = [&]( const int tur ) {
+        return part_flag( tur, "MANUAL" ) || parts[tur].mode == 0;
+    };
+    turrets.erase( std::remove_if( turrets.begin(), turrets.end(), cant_shoot ), turrets.end() );
+    if( turrets.empty() ) {
+        add_msg( m_warning, _("Can't aim turrets: all turrets are offline") );
```
Comment (kevingranade) [inline src/vehicle.cpp]: I'm probably just doing something wrong, but I can't trigger anything but this message, e.g. trying to invoke it from several different APCs.  Also I can't toggle the modes of the individual turrets, they're stuck in off I think, even though they say auto.
Does this only interact with a particular type of turret?

Comment (kevingranade) [inline src/vehicle.cpp]: OR!  Perhaps I forgot to merge the code before testing it <_<

**Comment by Coolthulhu** [comment]
OK, it looks like I broke the menu and didn't notice. Should be fixed now.
Also added more warnings for players, so that manual turret firing won't just silently bail out if something isn't right.

There is a minor issue with charger gun ranges (sometimes uses full range when in low-power mode), but it is really minor. The low power charger gun mode has damage so low that it can't really be exploited for anything. Will probably fix later, when I'll be trying to add aiming (as in: improving steadiness) to turrets.

**Comment by chaosvolt** [comment]
Highly belated question: Does a manually-mounted gun use the weapon's dispersion and recoil, or does it use build item volume like Blaze's turret notes claim? owo

**Comment by Coolthulhu** [comment]
It uses stats from the item used to construct the part.

**Comment by chaosvolt** [comment]
Ooh goody, so adding a higher-volume version of the 120mm gun (for an unreleased mod) won't ruin its accuracy? owo

**Pull request merged**